### DISCUSSION TITLE
Arc Commander - Authentication Step Fix - detailed version

### DISCUSSION THREAD

**Discussion by andreaschrader** [nfdi4plants/arcCommander#157] [Show and tell]
In our CEPLAS course in November 2022, a rare case of a problem with the authentication step occured (“arc remote accesstoken get -s https://git.nfdi4plants.org”).
Below you can find in German the quick DataPLANT response from @HLWeil and a longer detailed version I wrote in order to allow the participant also several days after the course to conduct the suggested steps. 
Maybe this is of help for others, so I left it here. German version first, **English version** below.

UPDATE (06-2023): A respective question and answer is now also included in the FAQ section of the knowledgebase. Please visit the [FAQs](https://nfdi4plants.org/nfdi4plants.knowledgebase/docs/faqs/faqs.html) for more FAQs on this and other DataPLANT tools.

UPDATE (07.2023): In case you create a ***project access token*** (“if you used the left sidebar settings/Access Token”) and not a ***personal access token***, a GitLab bot will join your project for this task. Please note that this is supportive and not a “negative” bot in case you are not familiar with this. You can read [here]( https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#bot-users-for-projects) more about this. (Thanks to @Zerskk and @j-bauer for providing the background knowledge on this.) However, for the steps below, you want to create a **personal access token**.

TLDR:
1.	GitLab Account haben
2.	GitLab Access Token erstellen (Bei Preferences -> Access Tokens) (Mit API und Read/write repository)
3.	Tokenstring kopieren
4.	Einmal ein GitLab Repo klonen mit angepasster Adresse (natürlich eins wo man Rechte drauf hat): git clone https://oauth2:TOKENSTRING@git.nfdi4plants.org/abc/xyz

Ausführliche Version:
-> Auf https://git.nfdi4plants.org/explore  gehen und „sign in“ durchführen

Am besten ein Projekt zum Testen einrichten:
-> Menu -> Projects -> Your projects
-> Create project
-> Create blank project 
-> Project name vergeben, z.B. “Test"
-> Visibility Level sollte “Private” sein
-> unten den Haken vor README entfernen
-> Create project

-> Oben rechts neben dem “Avatar” im Dropdown-Menü “Preferences” wählen.
-> Links in der Leiste dann “Access Tokens” wählen.
-> Token Namen vergeben, bei api, read_repository und write_repository einen Haken setzen.
-> Create personal access token

-> Token kopieren
-> cmd Befehl vorbereiten: git clone https://oauth2:TOKENSTRING@git.nfdi4plants.org/abc/xyz
-> Token bei TOKENSTRING einfügen
-> abc ist das GitHub handle (oben rechts, neben dem Avatar klicken, alle Zeichen nach dem “@“)
-> xyz der Name des Projektes, das gerade angelegt wurde
-> Im Explorer in einem Ordner gehen in den das Test Repositorium (oder hier “project” genannt) geklont werden soll. „cmd“ eingeben und den Befehl dort ausführen, alternativ im Command Prompt in diesen Ordner navigieren.

Soll das Test Repositorium gelöschen werden: im DataHUB in das Projekt gehen, dann links unter settings->general->advanced->Expand-> ganz unten “ Delete project”.

**Comment by andreaschrader** [comment]
English version of Arc Commander - Authentication Step Fix - detailed version #157

TLDR:
1.	Have a GitLab account
2.	Generate a GitLab access token (Preferences -> Access Tokens) (with API and read/write repository)
3.	Copy token string
4.	Clone a GitLab repo with adjusted addresse (one for which you have the respective access rights): git clone https://oauth2:TOKENSTRING@git.nfdi4plants.org/abc/xyz

Detailed version:

-> Go to https://git.nfdi4plants.org/explore  and sign in

Prepare one project for testing:
-> Menu -> Projects -> "Your projects" or "View all projects"
-> Create project
-> Create blank project 
-> You might need to select the user e.g. "mygitlabhandle"
-> Select a project name, e.g. “Test"
-> Visibility Level: “Private”
-> remove the tick in front of README
-> Create project

-> In the upper right corner, next to the avatar in the dropdown menu, select “Preferences”.
-> In the left menu, select “Access Tokens”.
-> Select a name fort he token, tick api, read_repository and write_repository.
-> Create personal access token

-> Copy token
-> Prepare cmd command: git clone https://oauth2:TOKENSTRING@git.nfdi4plants.org/abc/xyz
-> Replace „TOKENSTRING“ with the token.
-> abc is the GitHub handle (upper right corner, dropdown menu next to avatar, all parts of the string following “@“)
-> xyz is the name of the project just created
-> In Windows, go in the explorer in the directory in which the test repository shall be cloned (here named “project”)
enter „cmd“ and execute the command there, as an alternative, navigate in the Command Prompt to this directory.

In case you want to delete the Test repo again: navigate into the project within the DataHUB, on the lefthand side select settings and general 
->advanced->Expand-> at the bottom: “ Delete project”.
